### DISCUSSION TITLE
Handle `http.disconnect` in `FileResponse`

### DISCUSSION THREAD

**Discussion by jhominal** [encode/starlette#1726] [Ideas] [answered]
Hello,

After implementing #1715 I have been able to test `FileResponse` and I see that it does not handle a `http.disconnect` from `receive` - that is, it sends the whole file, regardless of whether the client is still connected or not.

Would you like a PR to add a test and implement that feature?

**Comment by Kludex** [comment, marked as answer]
https://github.com/encode/starlette/issues/1301#issuecomment-1028844042

Should work with: https://www.starlette.io/responses/#baizeasgifileresponse

**Reply by jhominal** [reply]
There is also @tomchristie's comment on the corresponding PR, which explains a bit more of the reasoning (which is to keep as many bells and whistles out of Starlette as possible).

https://github.com/encode/starlette/pull/1427#issuecomment-1028850195

**Comment by jhominal** [comment]
I guess that's a "no", then.

Would you like a PR for testing that `StreamingResponse` honors `http.disconnect` at least? I did not see any corresponding test in `test_responses.py` (and if you want to see the commit that implements both for `FileResponse` and the test, that is https://github.com/jhominal/starlette/commit/a129a79e5cce2360c8d30c75b7b3d94496622013)

**Reply by Kludex** [reply]
yes, please 🙏

**Reply by jhominal** [reply]
Done #1727 😄
